spec/support/shared/spree_builders.rb

```diff
@@ -32,12 +32,14 @@ def build_spree_product(id: 1, name: 'name', description: 'description',
   def build_spree_variant(product: nil, sku: 'sku',
                           weight: 10, weight_unit: 'oz',
                           price: 10.45, option_values: [],
+                          pos_variant_id: '321',
                           updated_at: build_date_time)
 
     variant = double(:spree_variant)
 
     allow(variant).to receive(:product).and_return(product)
     allow(variant).to receive(:product_id).and_return(product.id) if product.present?
+    allow(variant).to receive(:pos_variant_id).and_return(pos_variant_id)
```

Comment:
I wonder if this double is more or less performant over using a struct of some type. To me, and I could be very wrong but, all of these mocked interfaces against an ActiveRecord backed object is a smell.